GRPC  is not working?

I'd made different type of test without success. 
First connection type: Grpc without  TLS

```Java
RemoteGrpcServer gs = new RemoteGrpcServer("localhost", 50051,"root", "toortoor", false, List.of());    
RemoteGrpcDatabase tx = new RemoteGrpcDatabase(gs, "localhost", 50051, 2480, "ogm-test", "root", "toortoor");    
        
tx.begin(); 
``` 
throws 
```
Exception in thread "main" com.arcadedb.exception.TransactionException: Error on transaction begin
	at com.arcadedb.remote.grpc.RemoteGrpcDatabase.lambda$begin$0(RemoteGrpcDatabase.java:243)
	at com.arcadedb.remote.grpc.RemoteGrpcDatabase.callUnaryVoid(RemoteGrpcDatabase.java:2070)
	at com.arcadedb.remote.grpc.RemoteGrpcDatabase.begin(RemoteGrpcDatabase.java:234)
	at com.arcadedb.remote.RemoteDatabase.begin(RemoteDatabase.java:267)
	at test.DirectConnect.test(DirectConnect.java:108)
	at test.DirectConnect.<init>(DirectConnect.java:70)
	at test.DirectConnect.main(DirectConnect.java:914)
Caused by: io.grpc.StatusException: UNAVAILABLE: io exception
Channel Pipeline: [SslHandler#0, ProtocolNegotiators$ClientTlsHandler#0, WriteBufferingAndExceptionHandler#0, DefaultChannelPipeline$TailContext#0]
	at io.grpc.Status.asException(Status.java:547)
	at io.grpc.stub.ClientCalls.blockingV2UnaryCall(ClientCalls.java:198)
	at com.arcadedb.server.grpc.ArcadeDbServiceGrpc$ArcadeDbServiceBlockingV2Stub.beginTransaction(ArcadeDbServiceGrpc.java:845)
	at com.arcadedb.remote.grpc.RemoteGrpcDatabase.lambda$begin$0(RemoteGrpcDatabase.java:238)
	... 6 more
Caused by: io.grpc.netty.shaded.io.netty.handler.ssl.NotSslRecordException: not an SSL/TLS record: 00001204000000000000037fffffff000400100000000602000000000004080000000000000f00010000860700000000007fffffff00000001485454502f3220636c69656e74207072656661636520737472696e67206d697373696e67206f7220636f72727570742e204865782064756d7020666f722072656365697665642062797465733a20313630333031303064653031303030306461303330333038333864383338383964333761306165613666383934356134
	at io.grpc.netty.shaded.io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1353)
	at io.grpc.netty.shaded.io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1428)
```


Second try:
```Java
RemoteGrpcServer gs = new RemoteGrpcServer("localhost", 50051,"root", "toortoor", true, List.of());    
RemoteGrpcDatabase tx = new RemoteGrpcDatabase(gs, "localhost", 50051, 2480, "ogm-test", "root", "toortoor");    
        
 tx.begin();
        
 MutableVertex svt1 = tx.newVertex("SimpleVertexEx");
 svt1.set("svex", "svt1");
 svt1.save();
 MutableVertex svt2 = tx.newVertex("SimpleVertexEx");
 svt2.set("svex", "svt2");
 svt2.save();
 svt1.newEdge("SimpleVertexEx_ohmSVE", svt2);
 
 svt1.save();
 tx.commit();
``` 

Fail on the last save(). This test work fine with RemoteDatabase.

throws:
```
Exception in thread "main" com.arcadedb.remote.RemoteException: gRPC error: UpdateRecord: Transaction not active
	at com.arcadedb.remote.grpc.RemoteGrpcDatabase.handleGrpcException(RemoteGrpcDatabase.java:1918)
	at com.arcadedb.remote.grpc.RemoteGrpcDatabase.saveRecord(RemoteGrpcDatabase.java:674)
	at com.arcadedb.remote.RemoteMutableVertex.save(RemoteMutableVertex.java:63)
	at test.DirectConnect.test(DirectConnect.java:118)
	at test.DirectConnect.<init>(DirectConnect.java:70)
	at test.DirectConnect.main(DirectConnect.java:914)
Caused by: io.grpc.StatusException: INTERNAL: UpdateRecord: Transaction not active
	at io.grpc.Status.asException(Status.java:547)
	at io.grpc.stub.ClientCalls.blockingV2UnaryCall(ClientCalls.java:198)
	at com.arcadedb.server.grpc.ArcadeDbServiceGrpc$ArcadeDbServiceBlockingV2Stub.updateRecord(ArcadeDbServiceGrpc.java:793)
	at com.arcadedb.remote.grpc.RemoteGrpcDatabase.saveRecord(RemoteGrpcDatabase.java:665)
```

--- a/grpc-client/src/test/java/com/arcadedb/remote/grpc/RemoteGrpcDatabaseRegressionTest.java
+++ b/grpc-client/src/test/java/com/arcadedb/remote/grpc/RemoteGrpcDatabaseRegressionTest.java
@@ -22,6 +22,7 @@ import com.arcadedb.GlobalConfiguration;
 import com.arcadedb.graph.MutableVertex;
 import com.arcadedb.query.sql.executor.Result;
 import com.arcadedb.query.sql.executor.ResultSet;
+import com.arcadedb.remote.RemoteDatabase;
 import com.arcadedb.test.BaseGraphServerTest;
 import com.arcadedb.server.grpc.InsertOptions;
 import com.arcadedb.server.grpc.InsertOptions.ConflictMode;
@@ -34,6 +35,8 @@ import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
 
 import java.util.ArrayList;
 import java.util.Arrays;
@@ -41,6 +44,7 @@ import java.util.HashMap;
 import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.stream.Stream;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
@@ -68,6 +72,7 @@ class RemoteGrpcDatabaseRegressionTest extends BaseGraphServerTest {
 
   private RemoteGrpcServer   grpcServer;
   private RemoteGrpcDatabase grpc;
+  private RemoteDatabase     httpDb;
 
   @Override
   public void setTestConfiguration() {
@@ -98,6 +103,7 @@ class RemoteGrpcDatabaseRegressionTest extends BaseGraphServerTest {
   @BeforeEach
   void open() {
     grpc = new RemoteGrpcDatabase(this.grpcServer, "localhost", 50051, 2480, getDatabaseName(), "root", DEFAULT_PASSWORD_FOR_TESTS);
+    httpDb = new RemoteDatabase("localhost", 2480, getDatabaseName(), "root", DEFAULT_PASSWORD_FOR_TESTS);
 
     // Create isolated schema for these tests (id unique, name string, n integer)
     grpc.command("sql", "CREATE VERTEX TYPE `" + TYPE + "` IF NOT EXISTS", Map.of());
@@ -117,6 +123,14 @@ class RemoteGrpcDatabaseRegressionTest extends BaseGraphServerTest {
       } catch (Throwable ignore) {
       }
       grpc.close();
+    }
+    if (httpDb != null) {
+      try {
+        httpDb.rollback();
+      } catch (Throwable ignore) {
+      }
+      httpDb.close();
+      httpDb = null;
     }
   }
 
@@ -399,6 +413,107 @@ class RemoteGrpcDatabaseRegressionTest extends BaseGraphServerTest {
       }
     } finally {
       grpc.command("sql", "DELETE FROM `" + VERTEX_TYPE + "`", Map.of());
+    }
+  }
+
+  static Stream<String> allDatabases() {
+    return Stream.of("gRPC", "HTTP");
+  }
+
+  private RemoteDatabase resolveDatabase(final String label) {
+    return "gRPC".equals(label) ? grpc : httpDb;
+  }
+
+  @ParameterizedTest(name = "[{0}] newVertex().save() and newEdge() within a transaction should not throw 'Transaction not active'")
+  @MethodSource("allDatabases")
+  @DisplayName("Issue #3524: newVertex().save() and newEdge() within a transaction [parametric: gRPC + HTTP]")
+  void vertexSaveAndEdgeCreationWithinTransactionParametric(final String label) {
+    final RemoteDatabase db = resolveDatabase(label);
+    final String VERTEX_TYPE = "SVEx3524P";
+    final String EDGE_TYPE = "SVEx3524P_Edge";
+
+    db.command("sql", "CREATE VERTEX TYPE `" + VERTEX_TYPE + "` IF NOT EXISTS", Map.of());
+    db.command("sql", "CREATE PROPERTY `" + VERTEX_TYPE + "`.svex IF NOT EXISTS STRING", Map.of());
+    db.command("sql", "CREATE EDGE TYPE `" + EDGE_TYPE + "` IF NOT EXISTS", Map.of());
+
+    try {
+      db.begin();
+
+      MutableVertex svt1 = db.newVertex(VERTEX_TYPE);
+      svt1.set("svex", "svt1");
+      svt1.save();
+
+      MutableVertex svt2 = db.newVertex(VERTEX_TYPE);
+      svt2.set("svex", "svt2");
+      svt2.save();
+
+      // This should not throw 'Transaction not active'
+      svt1.newEdge(EDGE_TYPE, svt2);
+
+      // This save (updating the vertex after edge creation) should also work within the transaction
+      svt1.save();
+
+      db.commit();
+
+      // Verify both vertices exist
+      try (ResultSet rs = db.query("sql", "SELECT count(*) AS c FROM `" + VERTEX_TYPE + "`", Map.of())) {
+        assertThat(rs.hasNext()).isTrue();
+        Number count = rs.next().getProperty("c");
+        assertThat(count.longValue()).as("[%s] both vertices should be persisted", label).isEqualTo(2L);
+      }
+
+      // Verify the edge exists
+      try (ResultSet rs = db.query("sql", "SELECT count(*) AS c FROM `" + EDGE_TYPE + "`", Map.of())) {
+        assertThat(rs.hasNext()).isTrue();
+        Number count = rs.next().getProperty("c");
+        assertThat(count.longValue()).as("[%s] edge should be persisted", label).isEqualTo(1L);
+      }
+    } finally {
+      db.command("sql", "DELETE FROM `" + EDGE_TYPE + "`", Map.of());
+      db.command("sql", "DELETE FROM `" + VERTEX_TYPE + "`", Map.of());
+    }
+  }
+
+  @ParameterizedTest(name = "[{0}] newVertex().save() within a transaction should persist only on commit and rollback on abort")
+  @MethodSource("allDatabases")
+  @DisplayName("Issue #3524: newVertex().save() persist-only-on-commit semantics [parametric: gRPC + HTTP]")
+  void vertexSaveRollbackWithinTransactionParametric(final String label) {
+    final RemoteDatabase db = resolveDatabase(label);
+    final String VERTEX_TYPE = "SVEx3524PRoll";
+
+    db.command("sql", "CREATE VERTEX TYPE `" + VERTEX_TYPE + "` IF NOT EXISTS", Map.of());
+    db.command("sql", "CREATE PROPERTY `" + VERTEX_TYPE + "`.svex IF NOT EXISTS STRING", Map.of());
+
+    try {
+      // Verify rollback: creates a vertex and then rolls back
+      db.begin();
+
+      MutableVertex svt1 = db.newVertex(VERTEX_TYPE);
+      svt1.set("svex", "rollback1");
+      svt1.save();
+
+      db.rollback();
+
+      try (ResultSet rs = db.query("sql", "SELECT count(*) AS c FROM `" + VERTEX_TYPE + "`", Map.of())) {
+        Number count = rs.next().getProperty("c");
+        assertThat(count.longValue()).as("[%s] vertex should NOT be persisted after rollback", label).isEqualTo(0L);
+      }
+
+      // Verify commit: creates a vertex and commits
+      db.begin();
+
+      MutableVertex svt2 = db.newVertex(VERTEX_TYPE);
+      svt2.set("svex", "committed");
+      svt2.save();
+
+      db.commit();
+
+      try (ResultSet rs = db.query("sql", "SELECT count(*) AS c FROM `" + VERTEX_TYPE + "`", Map.of())) {
+        Number count = rs.next().getProperty("c");
+        assertThat(count.longValue()).as("[%s] vertex should be persisted after commit", label).isEqualTo(1L);
+      }
+    } finally {
+      db.command("sql", "DELETE FROM `" + VERTEX_TYPE + "`", Map.of());
     }
   }
 

